### PR TITLE
Sound Signals implementation

### PR DESCRIPTION
**Pull request by Chleba**
Added simple Sound class controling sound signals with additional C wrapper for old GUI.
Implementation is by the specification copying same behavior as MK3 firmware, so it have 4. modes and 5.types of signals.
BFW-408

### CODE REVIEW

File: src/gui/dialogs/DialogLoadUnload.cpp [hunk 1]
```diff
@@ -51,6 +52,11 @@ static const char *txt_purging            = "Purging";
 static const char *txt_is_color           = "Is color correct?";
 static const char *txt_nozzle_cold        = "Nozzle is too cold.";
 
+// Phase callbacks to play a sound in specific moment at the start/end of
+// specified phase
+void userPushEnter (){  Sound_DoSound(eSOUND_TYPE_StandardPrompt); }
```
Comment (RadekVana) [inline src/gui/dialogs/DialogLoadUnload.cpp:57]: should not it be member function?
void DialogLoadUnload::userPushEnter () ...

File: src/guiapi/src/window_frame.c [hunk 2]
```diff
@@ -36,18 +37,28 @@ void window_frame_event(window_frame_t *window, uint8_t event, void *param) {
     case WINDOW_EVENT_ENC_DN:
         dif = (int)param;
         id = window_focused();
-        while (dif--)
+        while (dif--) {
```
Comment (RadekVana) [inline src/guiapi/src/window_frame.c:40]: i think those brackets  `{` after `while (dif--)` are wrong
seems like 
```
if (id >= 0)
            window_set_focus(id);
```
was intended to run only once

File: src/gui/dialogs/DialogLoadUnload.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/dialogs/DialogLoadUnload.cpp:57]: also those methods should be outside clang-format off section

File: src/common/eeprom.h [hunk 3]
```diff
@@ -54,9 +54,12 @@
     #define EEVAR_LAN_HOSTNAME  0x1A // 20char string
 #endif                               // (EEPROM_FEATURES & EEPROM_FEATURE_LAN)
 
-#define EEVAR__PADDING 0x19 // 1..4 chars, to ensure (DATASIZE % 4 == 0)
+// sound variable
+#define EEVAR_SOUND_MODE 0x1B // uint8_t
```
Comment (RadekVana) [inline src/common/eeprom.h:58]: 0x1b

Comment (RadekVana) [reply src/common/eeprom.h:58]: also change please EEVAR_LAN_HOSTNAME  0x1A to 0x1a

File: src/guiapi/src/window_frame.c [hunk 4]
```diff
@@ -36,18 +37,28 @@ void window_frame_event(window_frame_t *window, uint8_t event, void *param) {
...
+                // End indicator of the frames list ->
+                Sound_DoSound(eSOUND_TYPE_BlindAlert);
+            }
+        }
         break;
     case WINDOW_EVENT_ENC_UP:
         dif = (int)param;
         id = window_focused();
-        while (dif--)
+        while (dif--) {
```
Comment (RadekVana) [inline src/guiapi/src/window_frame.c:53]: again brackets

File: src/guiapi/src/window_frame.c (hunk 2 above)
Comment (Chleba) [reply src/guiapi/src/window_frame.c:40]: my bad - sorry

File: src/guiapi/src/window_frame.c (hunk 4 above)
Comment (Chleba) [reply src/guiapi/src/window_frame.c:53]: again, my bad - sorry

File: src/guiapi/src/window_msgbox.c [hunk 5]
```diff
@@ -132,6 +133,7 @@ void window_msgbox_click(window_msgbox_t *window) {
     //int count = window_msgbox_button_count[btn]; // get number of buttons from table
     int idx = ((window->flags & MSGBOX_MSK_IDX) >> MSGBOX_SHI_IDX); // selected button index
     window->res = window_msgbox_buttons[btn][idx];
+    Sound_StopSound();
```
[the remark this answers is not in the document]
Comment (Chleba) [reply src/guiapi/src/window_msgbox.c]: right!

**Pull request merged**